pcd_to_gridmap:
  ros__parameters:
    pcd_path: "/home/cod-sentry/qza_ws/pcd2pgm/src/COD_PCD2PGM/map.pcd"
    pcd2gridmapConfig: "/home/cod-sentry/qza_ws/pcd2pgm/src/COD_PCD2PGM/config/PclLoaderParameters.yaml"
    map_frame_id: "map"
    thre_low: 0.1
    thre_high: 1.5
    is_negative: false
    radius: 5.0
    thre_count: 5.0
    filters:
      filter1:  # Fill holes in the map with inpainting.
        name: inpaint
        type: gridMapCv/InpaintFilter
        params:
          input_layer: elevation
          output_layer: elevation_inpainted
          radius: 0.1

      filter2:  # Compute surface normals.
        name: surface_normals
        type: gridMapFilters/NormalVectorsFilter
        params:
          input_layer: elevation_inpainted
          output_layers_prefix: normal_vectors_
          radius: 0.2
          normal_vector_positive_axis: z

      filter3:  # Compute slope from surface normal.
        name: slope
        type: gridMapFilters/MathExpressionFilter
        params:
          output_layer: slope
          expression: acos(normal_vectors_z)

